### PR TITLE
Document prow-tests:beta flow and how to use it

### LINKED ISSUE #1364: Staging mechanism for new prow-tests images
Problem: No good or documented way to test new prow-tests images.

Solution:
Building on https://github.com/knative/test-infra/issues/329, add an optional presubmit using an alpha tagged prow-tests image, change a CI flow to use a beta-tagged image, and document the procedure.

### PR DESCRIPTION
**Pull request by coryrc**
Fixes #1364

/cc @chizhg 
/cc @chaodaiG

### CODE REVIEW

File: images/prow-tests/README.md [hunk 1]
```diff
@@ -1,15 +1,43 @@
...
 
 ## Building and publishing a new image
 
 See parent README.md
+
+## Testing and Deploying Images
+
+When `make push` is run for `prow-tests`, it publishes your image with the
+`:beta` label. At night Pacific time, a set of duplicate jobs are run using
+this new image and viewable at https://testgrid.knative.dev/beta-prow-tests
```
Comment (mattmoor) [inline images/prow-tests/README.md:19]: Format markdown:
```suggestion
`:beta` label. At night Pacific time, a set of duplicate jobs are run using this
new image and viewable at https://testgrid.knative.dev/beta-prow-tests
```

File: images/prow-tests/README.md [hunk 2]
```diff
@@ -1,15 +1,43 @@
...
+
+A normal release process would involve first editing Dockerfile and/or
+programs it installs from test-infra, then running `make iterative-build` to
+build a local copy and ensure it still builds. Next, if adding a new tool or
+doing something complicated, do some rudimentary exploration in the image by
+running `make iterative-shell` to ensure it's working; most usage of prow-tests
+is via `runner.sh` so you might check a new version of Go by running
+`GO_VERSION=go1.30 runner.sh bash` and running `go version` or `go get something`.
```
Comment (mattmoor) [inline images/prow-tests/README.md:27]: Format markdown:
```suggestion
A normal release process would involve first editing Dockerfile and/or programs
it installs from test-infra, then running `make iterative-build` to build a
local copy and ensure it still builds. Next, if adding a new tool or doing
something complicated, do some rudimentary exploration in the image by running
`make iterative-shell` to ensure it's working; most usage of prow-tests is via
`runner.sh` so you might check a new version of Go by running
`GO_VERSION=go1.30 runner.sh bash` and running `go version` or
`go get something`.
```

File: images/prow-tests/README.md [hunk 3]
```diff
@@ -1,15 +1,43 @@
...
+
+With an image you feel comfortable with deploying, create a PR to knative/test-infra
+and get approval. Once merged, pull upstream into your master branch and run
+`make push`. This will upload your image to the registry at
```
Comment (mattmoor) [inline images/prow-tests/README.md:31]: Format markdown:
```suggestion
With an image you feel comfortable with deploying, create a PR to
knative/test-infra and get approval. Once merged, pull upstream into your master
branch and run `make push`. This will upload your image to the registry at
```

File: images/prow-tests/README.md [hunk 4]
```diff
@@ -1,15 +1,43 @@
...
+`gcr.io/knative-tests/test-infra/prow-tests` with a date-commit_hash, `latest`,
+and `beta` tags. Let the image run at least a single overnight and ensure the
+jobs in the https://testgrid.knative.dev/beta-prow-tests testgrid are as good
+as the jobs at https://testgrid.knative.dev/knative for master branch and
+releases under "knative-X.Y" at https://testgrid.knative.dev ; it's known that
+some are just bad so you will either learn to know them or just do a lot of clicking;
```
Comment (mattmoor) [inline images/prow-tests/README.md:37]: Format markdown:
```suggestion
jobs in the https://testgrid.knative.dev/beta-prow-tests testgrid are as good as
the jobs at https://testgrid.knative.dev/knative for master branch and releases
under "knative-X.Y" at https://testgrid.knative.dev ; it's known that some are
just bad so you will either learn to know them or just do a lot of clicking;
```

File: images/prow-tests/README.md [hunk 5]
```diff
@@ -1,15 +1,43 @@
...
+it's unlikely to take more than ten minutes to review all the jobs unless
+something is actually wrong.
+
+Once you are happy the :beta image is working, go to https://gcr.io/knative-tests/test-infra/prow-tests
+and add the label `stable` to your image with the `beta` tag. Monitor #test on
+Slack to see if any new complaints appear.
```
Comment (mattmoor) [inline images/prow-tests/README.md:43]: Format markdown:
```suggestion
Once you are happy the :beta image is working, go to
https://gcr.io/knative-tests/test-infra/prow-tests and add the label `stable` to
your image with the `beta` tag. Monitor #test on Slack to see if any new
complaints appear.
```

File: images/prow-tests/README.md [hunk 6]
```diff
@@ -1,15 +1,43 @@
...
+running `make iterative-shell` to ensure it's working; most usage of prow-tests
+is via `runner.sh` so you might check a new version of Go by running
+`GO_VERSION=go1.30 runner.sh bash` and running `go version` or `go get something`.
+
+With an image you feel comfortable with deploying, create a PR to knative/test-infra
+and get approval. Once merged, pull upstream into your master branch and run
+`make push`. This will upload your image to the registry at
+`gcr.io/knative-tests/test-infra/prow-tests` with a date-commit_hash, `latest`,
+and `beta` tags. Let the image run at least a single overnight and ensure the
+jobs in the https://testgrid.knative.dev/beta-prow-tests testgrid are as good
```
Comment (chaodaiG) [inline images/prow-tests/README.md:34]: Use hyperlink please

File: images/prow-tests/README.md (hunk 4 above)
Comment (chaodaiG) [inline images/prow-tests/README.md:37]: Ideally this should be an automated process

File: images/simple-image.mk [hunk 7]
```diff
@@ -54,4 +54,4 @@ push_latest: confirm-master build
 	docker tag $(IMG):$(TAG) $(IMG):latest
 	docker push $(IMG):latest
 
-push: push_versioned push_latest
+push:: push_versioned push_latest
```
Comment (chaodaiG) [inline images/simple-image.mk:57]: Curious why this change

File: images/prow-tests/README.md (hunk 6 above)
Comment (coryrc) [reply images/prow-tests/README.md:34]: for? gcr.io?

File: images/simple-image.mk (hunk 7 above)
Comment (coryrc) [reply images/simple-image.mk:57]: Needed so push:: in the prow-tests/Makefile can add a requirement.

File: images/prow-tests/README.md (hunk 6 above)
Comment (chaodaiG) [reply images/prow-tests/README.md:34]: gcr.io is an exception, the others could be hyperlinks

Comment (coryrc) [reply images/prow-tests/README.md]: I don't know what you are asking. All http[s]:// links become hyperlinks in Markdown.

File: images/prow-tests/README.md [hunk 8]
```diff
@@ -1,15 +1,45 @@
...
+`runner.sh` so you might check a new version of Go by running
+`GO_VERSION=go1.30 runner.sh bash` and running `go version` or
+`go get something`.
+
+With an image you feel comfortable with deploying, create a PR to
+knative/test-infra and get approval. Once merged, pull upstream into your master
+branch and run `make push`. This will upload your image to the registry at
+http://gcr.io/knative-tests/test-infra/prow-tests with a date-commit_hash,
+`latest`, and `beta` tags. Let the image run at least a single overnight and
+ensure the jobs in the https://testgrid.knative.dev/beta-prow-tests testgrid are
```
Comment (chaodaiG) [inline images/prow-tests/README.md:35]:
```suggestion
ensure the jobs in[testgrid](https://testgrid.knative.dev/beta-prow-tests) are
```

Here is an example

Comment (coryrc) [reply images/prow-tests/README.md:35]: There are multiple different testgrid dashboard groups being discussed and doing this adds about 50% more text when reading the markdown in the editor versus only removing 50% when viewing as a web page. That's why I'd prefer not to, but your call knowing my reasoning.

Comment (chaodaiG) [reply images/prow-tests/README.md:35]: We are certainly viewing this from different points of angles. I agree with you that adding these making code review harder, but this is one time thing vs. who knows how many times people go through this README, so I'm weighing more on the consumer side. 

However, this is not so big of a reason for blocking you, so 
/approve

Comment (coryrc) [reply images/prow-tests/README.md:35]: It's not about code review; when I'm editing Dockerfile I open the README.md in emacs at the same time. It highlights URLs so it looks good to me, so the `[text stuff]()` is just noise. Were I making a table of contents or if the particulars of the URLs weren't important, I'd definitely use the link bracket form. It just feels ambiguous here. I'll change if you want or I need /lgtm :-D
